### ISSUE TITLE
After hot restart on Android AppLovinMAX.initialize() never completes

### ISSUE THREAD

**Issue by eli1stark** [AppLovin/AppLovin-MAX-Flutter#212]
### MAX Plugin Version

N/A

### Flutter Version

3.9.2

### Device/Platform Info

Xiaomi MI9 Lite

### Current Behavior

When you call this code after hot restart on Android:

```dart
await AppLovinMAX.initialize(apiKey);
```

the `Future` never completes.

Works fine on the first launch.

### Expected Behavior

After hot restart on Android AppLovinMAX.initialize() should complete as normal.

### How to Reproduce

1. Call on Android:

    ```dart
    await AppLovinMAX.initialize(apiKey);
    ```

     It should complete as usual.

2. Now hot restart the Flutter app.

3. Call again:

    ```dart
    await AppLovinMAX.initialize(apiKey);
    ```

     initialize should never complete.

### Additional Info

It might be reproducible in other cases, hot restart is just the easiest one.

**Comment by eli1stark** [comment]
Okay, I think it might be a duplicate of https://github.com/AppLovin/AppLovin-MAX-Flutter/issues/210.

Can we document this in the initialize method? Something like this:

```dart
/// Initializes the SDK.
///
/// This method must be called only once. Calling it multiple times may cause the
/// returned future to never complete.
///
/// For more information, see the [Initialize the SDK](https://dash.applovin.com/documentation/mediation/flutter/getting-started/integration#initialize-the-sdk) guide.
Future<void> initialize(String sdkKey) async {
  // Implementation...
}
```

**Comment by eli1stark** [comment]
Created pr: https://github.com/AppLovin/AppLovin-MAX-Flutter/pull/213

**Comment by eli1stark** [comment]
Closing as duplicate

**Comment by eli1stark** [comment]
Actually, no, I don't think that using safeguards like `isInitialized` can help in this case because a hot restart resets all state, leaving the issue open just in case.

**Comment by kayces** [comment]
I encountered the same problem, and I found that it behaves differently in different projects. In mature projects, hot restart never returns, but in a minimal demo, it works fine.

**Comment by ycv005** [comment]
Yes, I am facing same issue. App got stuck in hot restart at app lovin initialization.

**Comment by ycv005** [comment]
Earlier, this was not happening and even, I am using 3.8.0, it is working fine, but 3.9.1 is causing issue. @thomasmso @eli1stark

**Comment by ycv005** [comment]
@thomasmso @alhiwatan Earlier it was working, but why new update has issue. LIke we are restarting application via restart still failing.

**Comment by alhiwatan** [comment]
@ycv005, we have just released [3.10.1](https://github.com/AppLovin/AppLovin-MAX-Flutter/releases/tag/release_3_10_1) that fixes the issue.  Please feel free to reopen if you still have the issue.

**Comment by ycv005** [comment]
@alhiwatan  - https://github.com/AppLovin/AppLovin-MAX-Flutter/issues/211#issuecomment-2183913640
